9 1 2

uzzelk opened the pull request.

alicampbell2 reviewed (commented):
## 9-1-2 Code review

#### Documentation

Any new code includes all the following forms of documentation:
  
- [x] README file for the indicator
- [x] Example input and output that run successfully locally.
  
If the code includes functions, or is a function/set of functions for SDGupdater:  
- [NA] **Function Documentation** as docstrings within the function definition.
- [X] **Examples** demonstrating major functionality, which runs successfully locally.

#### Functionality

- [see comments] **Installation**: Installation or build of the code succeeds.
- [x] *Guidelines**: New code conforms to the project contribution
  guidelines.  
- [x] **Example data**: Example output generated using the example input matches the given example output. 
- [x] **csv formatting**: The csv output is correct, given general guidance for csvs (e.g. columns are in 
a logical order, values within columns are in the correct order, sentence case is used, correct terms are 
used (i.e. ar in the data dictionary), observation status column is correctly filled in).  
     
If the code includes functions, or is a function/set of functions for SDGupdater:  
- [NA] **Automated tests**: Unit tests cover essential functions for a reasonable range
  of inputs and conditions. All tests pass on your local machine.


#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**


---

### Review comments

*Insert detailed comments here!*

- The setwd(indicator) on line 23 of update_indicator_main.R requires that your working directory already be set to sdg_data_updates,
otherwise comes up with this `Error in setwd(indicator) : cannot change working directory`
- I got this error: `Error in getChildlessNode(xml = workbookRelsXML, tag = "Relationship") : function 'Rcpp_precious_remove' 
not provided by package 'Rcpp'`, which comes up when you try to run get_type_1_data.
	- When this has happened to me before, I just ditched get_type_1_data and used readxl::read_excel() instead. 
	get_type_1_data is the only SDGupdater package that has given me this problem
	- When I tried to update/reinstall Rccp, I got `package ‘Rccp’ is not available (for R version 4.0.2)`. I tried updating to 4.1.3
	and updating 'Rccp'. It took a few rounds of doing this and restarting R before it worked. Presumably installing SDGupdater
	installs Rccp as part of it because:
		- When I added Rccp into the packages at the top pf compile_tables.R I get this warning:
		`Warning in install.packages: package ‘Rccp’ is not available as a binary package for this version of R (4.1.3)
		A version of this package for your version of R might be available elsewhere,
		see the ideas at https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages`
		- And then when calling the library: `Error in library("Rccp") : there is no package called ‘Rccp’`
		- But get_type_1_data still works
		- So think this may just be my computer, but maybe worth leaving a warning in the README
- I also got this error after the above was fixed:
`> source("compile_tables.R")
Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) : 
  there is no package called ‘fastmap’`, which was fixed by adding 'fastmap' to the packages and library calling at the top of compile_tables.R
- The observation status column differs between this csv 'Normal value' and the one in the in progress folder 'Definition differs'

uzzelk commented:
Thanks for reviewing this so quickly @alicampbell2 - I've addressed your comments in the following ways:

* I've included a troubleshooting section in the README that instructs the person what to do in the event they get the same errors you experienced (I left the get_type_1_data function in there and focused on addressing the SDGupdater Rccp issue, as the SDGupdater functions should be used across automations for consistency)
* I added 'fastmap' to the packages and library calling at the top of compile_tables.R (although not sure if this would be needed if the SDGupdater package installs properly)

In terms of the observation status column the 'Definition differs' in the in progress file is leftover from a long time ago and should be 'Normal value' (unless the source data highlights them as something different e.g., provisional, estimated). The Definition differs would be used to indicate slight deviations from the established methodology (footnote-type information) where these divergences do not imply a break in time series. There is [SDMX guidance](https://officenationalstatistics.sharepoint.com/:x:/r/sites/SusIneqSDG/_layouts/15/Doc.aspx?sourcedoc=%7B260B2723-2FCD-4335-B710-16FA26B3A035%7D&file=SDG_DSD_MATRIX.1.7.xlsm&_DSL=1&action=default&mobileredirect=true) on this on SharePoint.

Hopefully that is everything, but let me know if I've missed anything/you want to discuss/you think there are other changes needed 😄
